EIPS/eip-4762.md

```diff
@@ -181,29 +168,27 @@ For `i` in `0 ... (len(code)+30)//31`.
 For a transaction, make these access events:
 
 ```
-(tx.origin, 0, VERSION_LEAF_KEY)
-(tx.origin, 0, BALANCE_LEAF_KEY)
-(tx.origin, 0, NONCE_LEAF_KEY)
-(tx.origin, 0, CODE_SIZE_LEAF_KEY)
-(tx.origin, 0, CODE_KECCAK_LEAF_KEY)
-(tx.target, 0, VERSION_LEAF_KEY)
-(tx.target, 0, BALANCE_LEAF_KEY)
-(tx.target, 0, NONCE_LEAF_KEY)
-(tx.target, 0, CODE_SIZE_LEAF_KEY)
-(tx.target, 0, CODE_KECCAK_LEAF_KEY)
+(tx.origin, 0, BASIC_DATA_LEAF_KEY)
+(tx.origin, 0, CODEHASH_LEAF_KEY)
```

Comment:
If we have a tx (non-value-bearing) targeting a precompile or system contract, we don't need to touch BASIC_DATA. (i.e: stateless client knows the code-size/hash)

We can still argue that this rule should be avoided to avoid a border case. But considering we didn't do any similar simplification for precompiles/system contracts in other places in the EIP (i.e. we avoid adding in the witness for them as much as possible), then maybe not doing it here is actually _adding_ a border case (we have to remember this is an exception), not removing it.

No strong opinions, just surfacing this.
cc @gballet @g11tech @gabrocheleau @tanishqjasoria